crates/cheatcodes/src/inspector.rs

```diff
@@ -1550,8 +1550,10 @@ where {
             let prev = account.info.nonce;
             let nonce = prev.saturating_sub(1);
             account.info.nonce = nonce;
-            // NOTE(zk): We sync with the nonce changes to ensure that the nonce matches
-            foundry_zksync_core::cheatcodes::set_nonce(sender, U256::from(nonce), ecx_inner);
+            if self.use_zk_vm {
+                // NOTE(zk): We sync with the nonce changes to ensure that the nonce matches
+                foundry_zksync_core::cheatcodes::set_nonce(sender, U256::from(nonce), ecx_inner);
```

Comment:
How is this related to enabling the foundry tests?

Reply:
because before it was running on _all_ regardless if zkvm was enabled or not

Reply:
We were accessing the nonce_holder_Address even though we were in evm exclusively and that was making an extra storage access that change the dictionary used by the fuzz tester